onadata/apps/restservice/signals.py

```diff
@@ -20,7 +21,8 @@ def save_kpi_hook_service(sender, instance, **kwargs):
         # Only register the service if it hasn't been created yet.
         if kpi_hook_service is None:
             kpi_hook_service = RestService(
-                service_url="/assets/{}/hook-signal/".format(instance.id_string),
+                service_url=settings.KPI_HOOK_ENDPOINT_PATTERN.format(
+                    asset_uid=instance.id_string),
```

Comment:
OK, so even though we [know](https://github.com/kobotoolbox/tasks/issues/310#issuecomment-505517980) that `id_string` is not always equal to the Asset's `uid`, I'm guessing this isn't easy to fix because KC has no knowledge of the `uid`?

Reply:
@jnm: I don't think it's so complicated. `kpi`can send its `uid` instead of a boolean. 

https://github.com/kobotoolbox/kpi/blob/master/kpi/signals.py#L36-L43
https://github.com/kobotoolbox/kpi/blob/master/kpi/deployment_backends/kobocat_backend.py#L300-L303
https://github.com/kobotoolbox/kobocat/blob/master/onadata/apps/logger/models/xform.py#L98

Reply:
Now tracked at https://github.com/kobotoolbox/kobocat/issues/577